### PR TITLE
The Case for Blockchain for Machine Learning

### LINKED ISSUE #2788: Can Blockchain improve Machine Learning privacy?
### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

Can Blockchain improve Machine Learning privacy?

### Introduction paragraph (2-3 paragraphs):

Machine Learning algorithms have thrived in the era of big data. But to use that data safely and effectively, we have to take privacy into account when developing Machine Learning systems. This is especially the case when it comes to using sensitive data such as protected health records.
There has been previous attempts to solve this problem using [Federated Learning](https://www.section.io/engineering-education/secured-deeplearning-in-remote-devices/) and [Differential Privacy](https://www.section.io/engineering-education/understanding-differential-privacy/) but it turns out that these systems have a flaw which we'll discuss in the article. And that Blockchain technology might help us solve it.

### Key takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article? Use this area to get your ideas down on the bulk of your article or tutorial.

By the end of this tutorial, a reader should be able to:

1. Understand the flaws associated with Federated Learning
2. Understanding the flaws associated with Differential Privacy
3. Understand how Blockchain technology can help solve this issue and improve ML privacy.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

1. [Machine Learning in/for Blockchain: Future and Challenges](https://arxiv.org/abs/1909.06189)
2. [Efficient Privacy-Preserving Machine Learning for Blockchain Network](https://ieeexplore.ieee.org/document/8827509)
3. [AI and Blockchain: A Disruptive Integration](https://ieeexplore.ieee.org/document/8481263)
4. [Secured Deep Learning in Remote Devices](https://www.section.io/engineering-education/secured-deeplearning-in-remote-devices/)

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by wilkistermumbi**
This Fixes #2788

### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/can-blockchain-improve-machine-learning-privacy/index.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+The computing nodes then compete to add the next block to the blockchain by solving a complex math puzzle. The winning node collects all the gradients to update that global model which is stored in the next block along with all the training information which needs to be appended to the chain.
+
+In this framework, all the identities of the data sources are anonymized and all the weights of the model are encrypted using differential privacy. So, both the identity of the users and their data are private. Additionally, before the next block is added to the chain by the computing node, that node will first go through and make sure that the encrypted identities of the data sources and the gradients match with the historical record on the blockchain. If they don't match, then they would not be included in the update.
```
Comment (collins-a) [inline content/articles/can-blockchain-improve-machine-learning-privacy/index.md:80]:
```suggestion
In this framework, all the identities of the data sources are anonymized and all the weights of the model are encrypted using differential privacy. So, both the identity of the users and their data are private. Additionally, before the next block is added to the chain by the computing node, that node will first go through and make sure that the encrypted identities of the data sources and the gradients match with the historical record on the blockchain. If they don't match, then they will not be included in the update.
```

File: content/articles/can-blockchain-improve-machine-learning-privacy/index.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+#### Data privacy
+When it comes to data privacy, blockchain can be used to help secure data sharing, allowing people to gain ownership over their data and make data more accessible to researchers. For example, patients would be able to efficiently and securely share their medical records with a new doctor, or share anonymized versions of their records with researchers who are interested in investigating a particular condition. 
+
+You could use blockchain to sell your data using smart contracts which usually run on the Ethereum blockchain. These are programs that only execute when specific pre-conditions are met, and which store and maintain a record of those conditions, and the result of the program in the blockchain itself. You can use smart contracts to do anything such as pay someone when a product is sold, issuing a parking ticket, or analyzing a patient's encrypted data when the patient permits for that data to be analyzed.
```
Comment (collins-a) [inline content/articles/can-blockchain-improve-machine-learning-privacy/index.md:85]:
```suggestion
You could use blockchain to sell your data using smart contracts which usually run on the Ethereum blockchain. These are programs that only execute when specific pre-conditions are met, and which store and maintain a record of those conditions, and the result of the program in the blockchain itself. You can use smart contracts to do things such as pay someone when a product is sold, issuing a parking ticket, or analyzing a patient's encrypted data when the patient permits for that data to be analyzed.
```

**Review by collins-a — COMMENTED** [review]
Good job @wilkistermumbi, near-flawless article. A couple of minor changes and we're good to go.

Plus, I feel like the article builds up to blockchain for ML. As such, the title could be something along the lines of _The Case for Blockchain for Machine Learning_ as it shows the limitations of DL and FL and introduces blockchain.

File: content/articles/can-blockchain-improve-machine-learning-privacy/index.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+But what if you don't want to deal with protecting the data itself? 
+
+Then you might turn to Federated Learning instead.
+
+### Federated Learning
+Unlike DP, Federated Learning (FL) does not focus on changing the data itself, instead, it focuses on training an algorithm across multiple separate datasets without sharing the information between the datasets themselves. 
+
+In this type of learning, each person's data must remain on their device. You build the main model of the cloud and send it to each person's phone. The model is trained on their device and a copy of all the trained is transmitted back to the cloud to be compiled into one large model. Depending on your goals, you may send copies of the global model back to each user's device to incorporate it into their personalized models for better generalization.
```
[the remark this answers is not in the document]
Comment (wilkistermumbi) [reply content/articles/can-blockchain-improve-machine-learning-privacy/index.md:49]: Yes. That's right. The main model is first build on the cloud and then it's sent to each person's phone where training takes place.

**Comment by wilkistermumbi** [comment]
Hi @wilkistermumbi Thank you for the review and interesting suggestions. I'm done making the requested changes.

**Review by collins-a — APPROVED** [review]

**Review by louisefindlay23 — APPROVED** [review]
Great article, @wilkistermumbi. Final review complete. Be sure to go through my changes to inform future articles.

Thanks, @collins-a for peer reviewing.

**Pull request merged**